### ISSUE TITLE
pi_switch install issue

### ISSUE THREAD

**Issue by jwbel80** [lexruee/pi-switch-python#18]
sudo pip install pi_switch within the /home/pi/wiringPi directory is taking hours freezing my pi 3  I've tried 3 times it cleared back to command prompt only once and showed an error that there is an encoding issue.  I would provide the exact error but i'm an hour into attempting this install again.  I did look into using the method you recommended with the pure python but as I'm attempting to replicate your AWShome project I need the script for acquiring the codes sent by the remote control on the etekcity device.  any help you can provide is appreciated.  I already have amazon svcs with alexa and voice running on my pi.  I thought your project would be easy to follow and get going, but I'm hung up.  Thanks in advance

**Comment by lexruee** [comment]
@jwbel80 I'm not the author of the AWShome tutorial. However, I'm sorry to hear that you encountered these troubles. The main problem is that the raspberry pi needs a lot of time to compile the heavyweight Python Boost dependency. Not to mention enough swap space to compile this dependency. 

This is also the main reason why I gave up this project. One way solve this problem would be to rewrite the C++/Python wrapper around the rc-swirch C++ code using the lightweight pybind11 library.

**Comment by jwbel80** [comment]
Thanks for responding so quickly Alex I will migrate my image to a larger SD card and see if I can't get it to finish compiling I didn't look at my swap space so I don't didn't think to consider that


Sent from my Samsung Galaxy smartphone.
-------- Original message --------From: Alexander Rüedlinger <notifications@github.com> Date: 4/30/17  6:56 PM  (GMT-05:00) To: lexruee/pi-switch-python <pi-switch-python@noreply.github.com> Cc: jwbel80 <jwbel80@gmail.com>, Mention <mention@noreply.github.com> Subject: Re: [lexruee/pi-switch-python] pi_switch install issue (#18) 
@jwbel80 I'm not the author of the AWShome tutorial. However, I'm sorry to hear that you encountered these troubles. The main problem is that the raspberry pi needs a lot of time to compile the heavyweight Python Boost dependency. Not to mention enough swap space to compile this dependency.
This is also the main reason why I gave up this project. One way solve this problem would be to rewrite the C++/Python wrapper around the rc-swirch C++ code using the lightweight pybind11 library.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/lexruee/pi-switch-python","title":"lexruee/pi-switch-python","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/lexruee/pi-switch-python"}},"updates":{"snippets":[{"icon":"PERSON","message":"@lexruee in #18: @jwbel80 I'm not the author of the AWShome tutorial. However, I'm sorry to hear that you encountered these troubles. The main problem is that the raspberry pi needs a lot of time to compile the heavyweight Python Boost dependency. Not to mention enough swap space to compile this dependency. \r\n\r\nThis is also the main reason why I gave up this project. One way solve this problem would be to rewrite the C++/Python wrapper around the rc-swirch C++ code using the lightweight pybind11 library."}],"action":{"name":"View Issue","url":"https://github.com/lexruee/pi-switch-python/issues/18#issuecomment-298262886"}}}
